sql/src/main/java/org/elasticsearch/action/bulk/BulkShardProcessor.java

```diff
@@ -202,6 +209,7 @@ private ShardId shardId(String indexName, String id, @Nullable String routing) {
             ).shardId();
         } catch (IndexMissingException e) {
             if (!autoCreateIndices) {
+                indicesDeleted.add(indexName);
```

Comment:
Isn't there anyway a window for the race condition? Imho this is only "optimizing" the delete race condition to be less expensive. Not sure if this is worth it.